pyiceberg/io/pyarrow.py

```diff
@@ -535,6 +539,13 @@ def _initialize_azure_fs(self) -> FileSystem:
         if sas_token := self.properties.get(ADLS_SAS_TOKEN):
             client_kwargs["sas_token"] = sas_token
 
+        if client_id := self.properties.get(ADLS_CLIENT_ID):
+            client_kwargs["client_id"] = client_id
+        if client_secret := self.properties.get(ADLS_CLIENT_SECRET):
+            client_kwargs["client_secret"] = client_secret
+        if tenant_id := self.properties.get(ADLS_TENANT_ID):
+            client_kwargs["tenant_id"] = tenant_id
```

Comment:
The `client_id`, `client_secret` and `tenant_id` should be provided together, should we raise a `ValueError`? This way we hide Arrow exceptions where possible. 

<img width="679" height="217" alt="image" src="https://github.com/user-attachments/assets/c7e0466d-702d-4656-ac3b-55836c878487" />

Reply:
makes sense, added!